src/models/transfer.py

```diff
@@ -137,30 +156,34 @@ def add_slippage(self, slippage: SolverSlippage, log_saver: PrintStore) -> None:
 
     def merge(self, other: Transfer) -> Transfer:
         """
-        Merge two transfers (acts like addition)
-        if all fields except amount are equal, returns a transfer who amount is the sum
+        Merge two transfers (acts like addition) if all fields except amount are equal,
+        returns a transfer who amount is the sum.
+        Merges incur information loss (particularly in the category of redirects).
+        Note that two transfers of the same token with different receivers,
+        but redirected to the same address, will get merged and original receivers will be forgotten
         """
         merge_requirements = [
-            self.receiver == other.receiver,
+            self.recipient == other.recipient,
             self.token == other.token,
         ]
         if all(merge_requirements):
             return Transfer(
                 token=self.token,
-                receiver=self.receiver,
+                solver=self.recipient,
```

Comment:
This seems off. Solver should be the solver and not the potential redirect_target shouldn't it?

Reply:
The reason this looks off is because we have changed the field to "solver" (which is not natural for an object whose name is "Transfer").

You have two "transfers" that you are trying to merge. They can only be merged if they are for the same token and going to the same recipient. We have checked this condition and the resulting, merged, transfer it that which has the same token, recipient and whose amount is the sum of the two.